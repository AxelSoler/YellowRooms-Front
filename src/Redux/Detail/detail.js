import axios from "axios";

const GET_ROOMS = "details/rooms/GET_ROOMS";

export const getRoomsActionCreator = () => (dispatch) => {
  axios.get("http://localhost:3000/rooms").then((response) => {
    rooms = response.data;
    console.log(rooms);
    dispatch({ type: GET_ROOMS, payload: rooms });
  });
};
const roomsReducer = (state = [], action) => {
  switch (action.type) {
    case GET_ROOMS:
      return action.payload;
    default:
      return state;
  }
};

export default roomsReducer;
