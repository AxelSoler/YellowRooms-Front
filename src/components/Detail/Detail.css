.room {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.roomPicture {
  width: 80%;
}

.roomDetails {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 70%;
}

.roomName {
  font-size: 28px;
  width: fit-content;
}

.roomGuests {
  font-size: 16px;
  padding: 12px;
}

.roomDescription {
  font-size: 14px;
  padding: 2%;
  padding: 12px;
  background-color: #cececc;
}

.roomPrice {
  font-size: 24px;
  padding: 0 12px;
  display: flex;
  justify-content: space-between;
  border-bottom: 4px solid #ffde59;
}

.roomPrice span {
  width: fit-content;
  margin: 2% 0;
}

.roomStars {
  margin: 2% 0;
  width: fit-content;
}

.roomButton {
  margin-top: 10%;
  width: 100%;
}

.reserveBtn {
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: #fff;
  background-color: #91bd04;
  padding: 12px;
  border: 4px #737372;
  border-radius: 20px;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  font-size: 22px;
  cursor: pointer;
}

.iconBtn {
  width: fit-content;
}
