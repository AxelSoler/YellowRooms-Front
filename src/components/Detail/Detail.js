import React from 'react';
import { useSelector } from 'react-redux';
import { useParams, useNavigate } from 'react-router-dom';
import { IconContext } from 'react-icons';
import { MdOutlineHotel } from 'react-icons/md';
import { BsArrowRightCircle } from 'react-icons/bs';
import './Detail.css';

const Detail = () => {
  const navigator = useNavigate();
  const { id } = useParams();

  const room = useSelector((state) => state.rooms.rooms.find((room) => room.id === id));

  const { attributes } = room;
  return room ? (
    <div className="room">
      <img src={attributes.photo} alt="Room" className="roomPicture" />
      <aside className="roomDetails">
        <div className="roomName">
          {attributes.name}
        </div>
        <div className="roomStars">
          <span>Stars</span>
          <span>{attributes.stars}</span>
        </div>
        <div className="roomGuests">
          <span>Ideal for </span>
          <span>{attributes.persons_allowed}</span>
          <span> people</span>
        </div>
        <div className="roomDescription">
          <span>{attributes.description}</span>
        </div>
        <div className="roomPrice">
          <span>Price:</span>
          <span>
            $
            {' '}
            {attributes.price}
          </span>
        </div>
        <div className="roomButton">
          <button
            type="button"
            className="reserveBtn"
            onClick={() => navigator(`/Reserve/${room.id}`)}
          >
            <IconContext.Provider
              value={{ size: '24px' }}
            >
              <MdOutlineHotel />
            </IconContext.Provider>
            Reserve Room
            <IconContext.Provider
              value={{ size: '24px' }}
            >
              <BsArrowRightCircle />
            </IconContext.Provider>
          </button>
        </div>
      </aside>
    </div>
  ) : (
    // if no room is found with the given id
    <div>No room found</div>
  );
};

export default Detail;
