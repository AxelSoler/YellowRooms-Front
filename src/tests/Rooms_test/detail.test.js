import React from 'react';
import renderer from 'react-test-renderer';
import wrapper from '../wrapMeInValidComponents';
import Detail from '../../components/Detail/Detail';

test('Rooms show render the same', () => {
  const jsonRep = renderer.create(wrapper(<Detail />)).toJSON();
  expect(jsonRep).toMatchSnapshot();
});
