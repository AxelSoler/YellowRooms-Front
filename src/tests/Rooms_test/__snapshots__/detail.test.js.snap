// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms show render the same 1`] = `
Array [
  <div>
    No room found
  </div>,
  ";",
]
`;
